File: dodam-system-domain/dodam-domain-rds/src/main/java/b1nd/dodam/domain/rds/bus/service/BusService.java
package b1nd.dodam.domain.rds.bus.service;

import b1nd.dodam.domain.rds.bus.entity.Bus;
import b1nd.dodam.domain.rds.bus.exception.BusApplicantNotFoundException;
import b1nd.dodam.domain.rds.bus.repository.BusApplicantRepository;
import b1nd.dodam.domain.rds.bus.repository.BusRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class BusService {
    private final BusApplicantRepository busApplicantRepository;
    private final BusRepository busRepository;

    public List<Bus> getAll() {
        return busRepository.findAll();
    }

    public Bus getById(Long id) {
        return busRepository.findById(id)
            .orElseThrow(BusApplicantNotFoundException::new);
    }

    public void save(Bus bus) {
        busRepository.save(bus);
    }

    public void deleteById(Long busId) {
        busApplicantRepository.deleteByBus_Id(busId);
        busRepository.deleteById(busId);
    }
}

File: services/service-bus/src/main/kotlin/com/b1nd/dodamdodam/bus/domain/bus/service/BusService.kt
package com.b1nd.dodamdodam.bus.domain.bus.service

import com.b1nd.dodamdodam.bus.domain.bus.entity.BusEntity
import com.b1nd.dodamdodam.bus.domain.bus.exception.BusNotFoundException
import com.b1nd.dodamdodam.bus.domain.bus.repository.BusApplicantRepository
import com.b1nd.dodamdodam.bus.domain.bus.repository.BusRepository
import org.springframework.stereotype.Service

@Service
class BusService(
    private val busRepository: BusRepository,
    private val busApplicantRepository: BusApplicantRepository
) {
    fun getAll(): List<BusEntity> {
        return busRepository.findAll()
    }

    fun getById(id: Long): BusEntity {
        return busRepository.findById(id)
            .orElseThrow { BusNotFoundException() }
    }

    fun create(bus: BusEntity): BusEntity {
        return busRepository.save(bus)
    }

    fun deleteById(busId: Long) {
        busApplicantRepository.deleteAllByBusId(busId)
        busRepository.deleteById(busId)
    }
}
